src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -128,88 +133,240 @@ internal static partial class DependentTypeFinder
                 cancellationToken.ThrowIfCancellationRequested();
 
                 if (project.SupportsCompilation)
+                    await DescendInheritanceTreeInProjectAsync(project).ConfigureAwait(false);
```

Comment:
sibling functinos became nested, to avoid having to many redundant parameters to pass.  now, the only things passed are those that are specific to that subcall.  I like that pattern here as it makes it clearer what is shared vs what is unique.